core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -128,42 +173,9 @@ abstract class RawModule extends BaseModule {
     // Check to make sure that all ports can be named
     checkPorts()
 
-    // Now that elaboration is complete for this Module, we can finalize names
-    for (id <- getIds) {
-      id match {
-        case id: ModuleClone[_]   => id.setRefAndPortsRef(_namespace) // special handling
-        case id: InstanceClone[_] => id.setAsInstanceRef()
-        case id: BaseModule       => id.forceName(default = id.desiredName, _namespace)
-        case id: MemBase[_]       => id.forceName(default = "MEM", _namespace)
-        case id: stop.Stop        => id.forceName(default = "stop", _namespace)
-        case id: assert.Assert    => id.forceName(default = "assert", _namespace)
-        case id: assume.Assume    => id.forceName(default = "assume", _namespace)
-        case id: cover.Cover      => id.forceName(default = "cover", _namespace)
-        case id: printf.Printf => id.forceName(default = "printf", _namespace)
-        case id: DynamicObject => {
-          // Force name of the DynamicObject, and set its Property[ClassType] type's ref to the DynamicObject.
-          // The type's ref can't be set upon instantiation, because the DynamicObject hasn't been named yet.
-          id.forceName(default = "_object", _namespace)
-          id.getReference.setRef(id.getRef)
-        }
-        case id: Data =>
-          if (id.isSynthesizable) {
-            id.topBinding match {
-              case OpBinding(_, _) =>
-                id.forceName(default = "_T", _namespace)
-              case MemoryPortBinding(_, _) =>
-                id.forceName(default = "MPORT", _namespace)
-              case PortBinding(_) =>
-                id.forceName(default = "PORT", _namespace, true, x => ModuleIO(this, x))
-              case RegBinding(_, _) =>
-                id.forceName(default = "REG", _namespace)
-              case WireBinding(_, _) =>
-                id.forceName(default = "_WIRE", _namespace)
-              // probes have their refs set eagerly
-              case _ => // don't name literals
-            }
-          } // else, don't name unbound types
-      }
+    // Take a second pass through any ids generated during atModuleBodyEnd blocks to finalize names for them.
+    for (id <- _ids.drop(numInitialIds)) {
+      nameId(id)
```

Comment:
```suggestion
    for (id <- _ids.view.drop(numInitialIds)) {
      nameId(id)
```
This probably won't matter much in practice, but that `.drop` does a full copy of the `ArrayBuffer`, whereas first asking for a `.view` then makes the `drop` lazy such that it won't materialize a new datastructure and instead will just let us iterate.

I experimented with a few alternatives likes so:
```scala
//> using scala "2.13.11"
//> using options "-unchecked", "-deprecation", "-language:reflectiveCalls", "-feature", "-Xcheckinit", "-Xfatal-warnings", "-Ywarn-dead-code", "-Ywarn-unused", "-Ymacro-annotations"

import scala.collection.mutable

object logEvent {
  private val totals = new mutable.HashMap[String, Double]()
  def apply[R](name: String)(block: => R): R = {
    val t0 = System.nanoTime()
    val result = block
    val t1 = System.nanoTime()
    val timeMillis = (t1 - t0) / 1000000.0

    val total = timeMillis + totals.getOrElse(name, 0.0)
    totals(name) = total
    println(f"$name took $timeMillis%.1f ms (total $total%.1f ms)")
    result
  }
}

object Main extends App {
  val array = logEvent("Make ArrayBuffer") {
    val xs = mutable.ArrayBuffer.empty[String]
    for (i <- 0 until 100000000) {
      xs += s"value_$i"
    }
    xs
  }

  for (_ <- 0 until 10) {

    logEvent(".drop") {
      var i = 0L
      for (x <- array.drop(array.size - 200000)) {
        i += x.size
      }
      println(i)
    }

    logEvent(".view.drop") {
      var i = 0L
      for (x <- array.view.drop(array.size - 200000)) {
        i += x.size
      }
      println(i)
    }

    logEvent(".iterator.drop") {
      var i = 0L
      for (x <- array.iterator.drop(array.size - 200000)) {
        i += x.size
      }
      println(i)
    }

    logEvent("using indexing") {
      var i = 0L
      var idx = array.size - 200000
      while (idx < array.size) {
        i += array(idx).size
        idx += 1
      }
      println(i)
    }
  }
}
```

Reply:
ah, thanks for the experimentation and pointers! btw, is there any canonical way to know this without reading the source or doing experiments? i didn't see any mention of these concerns here: https://www.scala-lang.org/api/2.13.3/scala/collection/mutable/ArrayBuffer.html#drop(n:Int):C